superset-frontend/src/components/DynamicEditableTitle/index.tsx

```diff
@@ -113,10 +113,12 @@ export const DynamicEditableTitle = ({
   // then we can measure the width of that span to resize the input element
   useLayoutEffect(() => {
     if (sizerRef?.current) {
-      sizerRef.current.innerHTML = (currentTitle || placeholder).replace(
-        /\s/g,
-        '&nbsp;',
-      );
+      sizerRef.current.innerHTML = (currentTitle || placeholder)
```

Comment:
@nytai are there libraries we can leverage for sanitizing the HTML?

Reply:
do we want to sanitize (ie, remove) html or just escape it? I think the purpose of this code is to render the title to the DOM so we can measure how much space it takes up, so we probably want to escape it. 

Looking at an escape-html lib, it's pretty basic. Eg https://github.com/component/escape-html/blob/master/index.js 

Happy to update the PR to use this lib if we think it's best, but... you know the joke about node modules 

![image](https://github.com/apache/superset/assets/10255196/65717c9c-ae6b-4a3c-8d8e-5575a5956da2)

Reply:
Sorry I meant escape rather than sanitize. It [seems](https://github.com/search?q=repo%3Aapache%2Fsuperset%20escape-html&type=code) like we're already installing the `escape-html` library and thus maybe we should just use that directly.

Reply:
That's the websocket sidecar app (for async queries), but sure we can use that.